Add community processes 2.0: (#43)
thegooddocsproject/thegooddocsproject.github.io: roles.md

layout: default
---

# Roles

Here we define roles within The Good Docs Project, along with the eligibility, expectations, and responsibilities associated with these roles.

**Version:** 2.0

**Last updated:** August 2020

## Community

Everyone contributing to, or using artifacts from, The Good Docs Project is considered part of our community. We encourage all community members to actively participate in our activities. This includes:

*   Participate in our forums and vote on motions.

> *   Raise and comment on bugs.

Reviewer: Raise and Comment on Issues

Author: Thanks, fixed.

> *   Suggest fixes by raising pull requests.

Reviewer 2: Comment and review PRs?

Author: Thanks, added.

**Access:**

> *   Community members are granted _read_ access to our github repositories.

Reviewer: this means we are sending them an invitation to joint the org?  They will have `read` access to all public repos by default.  Fine with sending them this invite, but think we need some mechanisms here.... Essentially, new community members join our organization should complete some onboarding potentially?

Author: I've tweaked wording based on your comment:
*   Community members have _read_ access to our github repositories (by default).

I don't think we need to introduce an onboarding process, at least not yet.

**Expectations of community members:**

> *   Community members are expected to be respectful to each other, in line with our [Code of Conduct](https://github.com/thegooddocsproject/governance/blob/master/CodeOfConduct.md).

Reviewer: Think it makes sense we create a set of onboarding materials that we expect them to complete.  have thoughts here.

Author: My initial feeling is that onboarding material would introduce a barrier to entry. But I suspect you have ideas I haven't considered. I suggest taking this to our forum for discussion, and potentially raise a bug to track it.

## Committers

A committer is a trusted community member who has additionally been given _write_ access to some or all of our github repositories.

**Access:**

*   Committers are granted _triage_, _maintain_ and _write_ access to the github repositories which they have contributed to.

**Additional expectations of committers:**

*   Understand and apply our quality guidelines.
*   Understand and abide by our license requirements.
*   Periodically contribute to our project over a one year period.

**Eligibility:**

Before being accepted for this role, a person should display the following characteristics:

*   Provide sustained, valuable contributions to the project, typically for three months or more. An example demonstration of this could be:

>     *   Submit pull requests which have been accepted and merged by other committers. Typically one large, three medium, or five small pull requests over three months.

Reviewer 2: Do we need to define what large, medium, and small PRs are? I guess small is typos, and large is structural changes, but not sure what medium might be.

Reviewer 3: Definitely something to capture as part of iteration 2 (or even I3) of this document.

Reviewer: Also, track people's contributions from the discussion side -- opening/commenting on  `Issues` and on PRs.  As not every contributor is going to be able to open a PR.  This means we aren't inclusive of less-experience individuals looking to become members.... Not sure if this means we change this from `Committers` to `Contributors`...

Author: Raised #47 to track this for a future version.

    *   Or provide equivalent contributions to our community, such as outreach, contributing to our forums, or coordinating activities. 
*   Express an interest in ongoing participation.
*   Acknowledge the committer expectations.

## Project Steering Committee

The Project Steering Committee (PSC) is the official managing body for our project. The committee is drawn from the community and is based on merit and interest.

*   We aim to attract a diverse mix of personal characteristics, reflective of the community we represent.

> *   We wish to avoid having one organisation employing a controlling proportion of the PSC.

Reviewer 2: ```suggestion
*   We wish to avoid having employees of any one organisation representing a controlling proportion of the PSC.
```

A PSC chair shall be selected from the PSC, with the only official extra duty being that they have a deciding vote in the case of a tie.

**Access:**

*   PSC members may be granted admin access to our github repositories if they have a need for them.

**Additional expectations of PSC members:**

*   Consider opinions from within our community, and advocate for the best interests of our community when making decisions.
*   Vote according to your personal best judgment, as opposed to voting on behalf of your employer.
*   Vote promptly (usually within days) to motions being raised.
*   Regularly participate in project activities.

> *   Help with whatever unglamourous work that is required to keep the project moving forward.

Reviewer 2: Personally, I like the term "plumbing" to describe this kind of work. I'm not sure any of what we do is particularly glamorous ;) (Also: needs to be corrected to US spelling)

Reviewer: Fine with this being here for now, but feel like we should have a larger `Core Values` statement documented which would be inclusive of this idea.  And those that don't embody those values probably shouldn't be members of the PSC.

Author: Tracked in #47

*   Monitor conversations within our forums and contribute constructively where appropriate.

> *   Optionally attend weekly meetings.

Reviewer 2: Really truly optional? If a PSC member never or rarely attended, wouldn't that be a problem?

Reviewer 3: Potentially yes. But in the past, it has more been radio silence on many fronts before we vote to remove a PSC member from the project.

Author: Because we are volunteers, who often can only fit open source work into an already busy day, and we are spread in timezones across the world, there will be valuable PSC members are can contribute constructively but who can't make it to the meeting timeslots selected.

I think that is okay. 

We can revisit this later if it becomes an issue.

I'm going to close this out with no action, but happy for you to add a comment to #47 or raise a new issue.

**Eligibility:**

Before being accepted for this role, a person should display the following characteristics:

*   Sustained track record displaying the characteristics expected of PSC members, typically while acting as a committer within our project, typically for six months or more.
*   Express an interest in ongoing participation.
*   Acknowledge the PSC member expectations.

## Mentors

Initiatives which are funded by The Good Docs Project should have a mentor.  

**Expectations of mentors:**

A mentor’s responsibilities typically includes:

*   Mentor a worker in developing a proposal.

> *   Facilitate making connections with other members within the community.

Reviewer: Feel like this is a two-way street.  Pairing Mentors <> Mentees can be a challenge, we should put for effort internally to help this happen as it won't develop organically.  Think we should create some sort of `team` or `members` structure inside of the good-docs github org.  And in it list the members and their roles.  and surface those that are mentors and available for mentoring.

Reviewer: Pair this into the onboarding

Author: Interesting suggestion. I'm a little wary of putting too much process over something which is often very organic. But this is worth discussing in our forums and adjusting a future version of our processes.
I'll close this with no action at the moment, but happy for you to raise as a bug to track it.

*   Attend weekly meetings and provide guidance where applicable.
*   If needed, escalate concerns to the Project Steering Committee.
*   Review work done, and recommend the approval or rejection of the project based on work done.

**Eligibility:**

A mentor will typically be a member of the PSC, but may be an experienced person who is trusted by the PSC.

Any conflict-of-interest the mentor may have should be stated beforehand. This could result from mentoring managers, direct reports, relatives, or close friends. The PSC will need to decide whether the conflict will negatively impact the mentoring experience.

## Treasurers

Treasurers are PSC members who additionally have been provided with admin access to our [financial account with Open Collective](https://opencollective.com/thegooddocsproject).

**Expectations of treasurers:**

*   Notify the PSC of Open Collective funding requests by [raising a motion](decisions.html).

> *   AFTER receiving approval from the PSC, approve funding requests.

Reviewer 2: Responsible financial management?

Reviewer 3: Put that in as an alpha cut, then plan to refine the sentence/guidance in Alpha 0.1 or 0.2?

Author: @Loquacity I'm unclear on what your suggested alternative is here. I've added your comment into #47.
I'm proposing this comment is not a blocker, and can get addressed in a future version.

**Eligibility:**

In order to limit the risk of misappropriation of funds, financial approval privileges should only be provided to a few long serving PSC members.

# Selection of roles

*   The PSC will review roles at least annually, in March, with the aims of:
    *   Inviting active and eligible community members into relevant roles, based on our eligibility criteria.
    *   Moving people who have been inactive for a year into a retired status, with thanks.
    *   Selecting a PSC chair.
*   Ad-hoc invitations may be initiated throughout the year, as needed.
*   Retired members should expect to be welcomed back if they become active again.

> *   PSC discussions about membership will typically be conducted privately in order to respect people’s feelings.

Reviewer 2: ```suggestion
*   PSC discussions about membership will typically be conducted privately in order to respect the feelings of those being discussed.
```
